### PR TITLE
Moved tqdm to get rid of issues with pickling

### PR DESCRIPTION
**Pull request by bonham79**
Fixes pickling issue by moving tqdm bars around. Actually just gets rid of them in favor of the loop wrapper. Moves along https://github.com/CUNY-CL/yoyodyne/pull/197

Also added flag to allow the em loop to skip validation (not really necessary to monitor unless you want to add a patience function in there).

### CODE REVIEW

File: maxwell/sed.py [hunk 1]
```diff
@@ -181,6 +178,7 @@ def fit_from_data(
         lines: Iterable[Tuple[Any]],
         copy_probability: Optional[float] = None,
         epochs: int = 10,
+        validate: bool = True,
```
[the remark this answers is not in the document]
Comment (bonham79) [reply maxwell/sed.py:181]: Oh I just left it as True so there was no change in previous behavior. False is my preferred.

**Comment by bonham79** [comment]
> This is a necessary fix but it seems to revert a bit of other progress. To wit:
> 
> * The custom formatting of the bar was nice, I think, and made the bars look like the ones in Yoyodyne. That's dropped here, I believe?

All that formatting is available for the wrapper if needed. Unsure if customizability is that important given that this is a lightweight library to offset Yoyodyne dependencies.

> * The previous design was modular and honestly read pretty clearly; this pushes the whole thing into a "God method" that is by all accounts less clear. To make this more concrete I think you should restore the `log_likelihood` method and use it inside the `em` method, which will make the latter easier to read. Is there some technical reason why this is not possible?

Log_likelihood is only ever called by `em` and even that is only occasional given the new bool. And it's only like, four locs. So the readibility is only so strong as an argument. Leaving it as its own method meant I needed to keep tqdm in another method. I'd rather have all the tqdm in one method so formatting fixes can be targeted to only the one method. So makes more sense to me to just merge the two.

**Comment by kylebgorman** [comment]
> > This is a necessary fix but it seems to revert a bit of other progress. To wit:
> > 
> > * The custom formatting of the bar was nice, I think, and made the bars look like the ones in Yoyodyne. That's dropped here, I believe?
> 
> All that formatting is available for the wrapper if needed. Unsure if customizability is that important given that this is a lightweight library to offset Yoyodyne dependencies.

I spent a lot of time figuring out how to do it...would you consider using it?

**Comment by bonham79** [comment]
> 
> I spent a lot of time figuring out how to do it...would you consider using it?

I'll toy around with adding back in, but push comes to shove I'm going to go for the wrapper instead of including a whole set of novel methods just for progress bars. That fair?

**Comment by kylebgorman** [comment]
> > I spent a lot of time figuring out how to do it...would you consider using it?
> 
> I'll toy around with adding back in, but push comes to shove I'm going to go for the wrapper instead of including a whole set of novel methods just for progress bars. That fair?

Sure; I asked because I think it's just an argument that has to be passed to the `tqdm` object you're creating.

**Comment by bonham79** [comment]
@kylebgorman this takes care of progress bar preferences?

File: maxwell/util.py [hunk 2]
```diff
@@ -2,6 +2,62 @@
...
+        "[{elapsed}<{remaining}, {rate_noinv_fmt}{postfix}]"
+    )
+
+    def __init__(self, total: int):
+        super().__init__(
+            total=total, position=0, leave=False, bar_format=self.BAR_FORMAT
+        )
+
+    def on_epoch_start(self) -> None:
+        raise NotImplementedError()
```
Comment (kylebgorman) [inline maxwell/util.py:20]: I think you don’t need () here

**Review by kylebgorman — APPROVED** [review]
One nit, otherwise looks good to me. Should I merge?

**Review by kylebgorman — COMMENTED** [review]
One nit

**Comment by tbartley94** [comment]
> One nit, otherwise looks good to me. Should I merge?

Yes please, helps cascade other PRs to done.

**Review by kylebgorman — APPROVED** [review]
LGTM

**Comment by bonham79** [comment]
@kylebgorman what's with the circle ci fail?

**Comment by bonham79** [comment]
Oh, unit tests. Aight i'll fix those.

**Pull request merged**